src/magic_folder/magic_folder.py

```diff
@@ -298,21 +302,36 @@ def from_config(cls, reactor, tahoe_client, name, config):
             Node(tahoe_client, tahoe_uri_from_string(mf_config.collective_dircap)),
             Node(tahoe_client, tahoe_uri_from_string(mf_config.upload_dircap)),
         )
+
         return cls(
             client=tahoe_client,
             config=mf_config,
             name=name,
+            local_snapshot_service=LocalSnapshotService(
+                mf_config.magic_path,
+                LocalSnapshotCreator(
+                    mf_config,
+                    mf_config.author,
+                    mf_config.stash_path,
+                ),
+            ),
             initial_participants=initial_participants,
-            _clock=reactor,
+            clock=reactor,
         )
 
-    def __init__(self, client, config, name, initial_participants, _clock=None):
-        super(MagicFolder, self).__init__()
+    @property
+    def name(self):
         # this is used by 'service' things and must be unique in this Service hierarchy
-        self.name = u"magic-folder-{}".format(name)
-        self._clock = _clock or reactor
+        return u"magic-folder-{}".format(self.folder_name)
+
+    def __init__(self, client, config, name, local_snapshot_service, initial_participants, clock):
+        super(MagicFolder, self).__init__()
+        self.folder_name = name
+        self._clock = clock
         self._config = config  # a MagicFolderConfig instance
         self._participants = initial_participants
+        self.local_snapshot_service = local_snapshot_service
+        local_snapshot_service.setServiceParent(self)
```

Comment:
don't we also need to call `startService()` on `local_snapshot_service`?

Reply:
>  don't we also need to call startService() on local_snapshot_service?

Looks like we don't need to. Starting the MagicFolder service seem to start the children services.